lib/blockscore/collection.rb

```diff
@@ -133,8 +133,7 @@ def create(params = {})
     # @api public
     def retrieve(id)
       each do |item|
-        next unless item.id == id
-        return item
+        return item if item.id.eql?(id)
```

Comment:
Yes, `#eql?` is the best you can get in this case if the `id` are string objects.